### ISSUE TITLE
Some emoji are not printed

### ISSUE THREAD

**Issue by aried3r** [hidakatsuya/prawn-emoji#28]
Hey! Thanks for this gem!

I was just playing around with this with different emoji and fonts and found some behavior which I don't know if it is a bug or I'm missing a configuration option.

```Ruby
Prawn::Document.generate 'sushi.pdf' do
  font 'DejaVuSans.ttf'
  text '👁🤷‍♀️🦄'
end
```

produces:

<img width="169" alt="Screenshot 2020-02-20 at 11 10 44" src="https://user-images.githubusercontent.com/1301152/74924143-3b383b80-53d2-11ea-94ef-bcb9e752935c.png">

The eye is part of [Unicode 7](https://emojipedia.org/eye/).

I tested with [unicode-emoji](https://github.com/janlelis/unicode-emoji) and maybe it's because it has a text representation?

```Ruby
» "👁🤷‍♀️🦄".scan(Unicode::Emoji::REGEX_INCLUDE_TEXT)XT)
=> [
  [0] "👁",
  [1] "🤷‍♀️",
  [2] "🦄"
]

» "👁🤷‍♀️🦄".scan(Unicode::Emoji::REGEX)
=> [
  [0] "🤷‍♀️",
  [1] "🦄"
]
```

Regarding unicode-emoji, is there a reason why the version constraint even contains the patch version? That way updates from 2.3 to 2.4 depend on a new release of this gem.

https://github.com/hidakatsuya/prawn-emoji/blob/26f83a1c31c70642291d3b2d4a2ddff6379dbe57/prawn-emoji.gemspec#L23

**Comment by hidakatsuya** [comment]
Thank you for your reporting.

:eye: is a Textual Emoji. 
https://emojipedia.org/emoji/%F0%9F%91%81/

Currently prawn-emoji does not render it as an emoji. And Its behavior depends on the following code:
https://github.com/hidakatsuya/prawn-emoji/blob/26f83a1c31c70642291d3b2d4a2ddff6379dbe57/lib/prawn/emoji.rb#L9

`Unicode::Emoji::REGEX_VALID` means:
> Matches (non-textual) singleton Emoji (except for singleton components, like a skin tone modifier without base Emoji) and all kind of valid Emoji sequences
> https://github.com/janlelis/unicode-emoji#main-regexes

From the above, the behavior that the emoji is not drawn is the expected but not a bug.

For example, I'm considering providing a feature to change its behavior as follows:
```ruby
require 'prawn/emoji'

Prawn::Emoji.matcher_regex = ::Unicode::Emoji::REGEX_INCLUDE_TEXT
```

> Regarding unicode-emoji, is there a reason why the version constraint even contains the patch version? That way updates from 2.3 to 2.4 depend on a new release of this gem.

Yes, there is. The reason is that the emoji version of both the emoji image (Twemoji) and the emoji matcher (unicode-emoji) always match.

**Comment by aried3r** [comment]
Thank you for the detailed reply, that's what I suspected. Can't promise I'll find the time for a PR. But if so, I'd guess we should replace `REGEX` with `self.regex`? A bit like the groupdate gem does it for [certain settings](https://github.com/ankane/groupdate/blob/96f07fe0c35e6fd0e953dd14e2b188eaccc35a34/lib/groupdate.rb#L17-L20).

**Comment by hidakatsuya** [comment]
@aried3r 
> so, I'd guess we should replace `REGEX` with `self.regex`? A bit like the groupdate gem does it for [certain settings](https://github.com/ankane/groupdate/blob/96f07fe0c35e6fd0e953dd14e2b188eaccc35a34/lib/groupdate.rb#L17-L20).

Sounds good, but the implementation requires activesupport. I implemented in #30 without activesupport. Could you review it?

**Comment by hidakatsuya** [comment]
Fixed by #30 and automatically closed. Please feel free to reopen if you'd like.

**Comment by aried3r** [comment]
Hey! Thank you for #30 and #31, I'm looking forward to trying it out!
